src/layout/mod.rs

```diff
@@ -1264,6 +1264,43 @@ impl<W: LayoutElement> Layout<W> {
         monitor.focus_column_left_or_last();
     }
 
+    pub fn focus_window_up_or_output(&mut self, output: &Output) -> bool {
+        if let Some(monitor) = self.active_monitor() {
+            let workspace = monitor.active_workspace();
+
+            if !workspace.columns.is_empty() {
+                let curr_idx = workspace.columns[workspace.active_column_idx].active_tile_idx;
+                let new_idx = curr_idx.saturating_sub(1);
+                if curr_idx != new_idx {
+                    workspace.focus_up();
+                    return false;
+                }
+            }
+        }
+
+        self.focus_output(output);
+        true
+    }
+
+    pub fn focus_window_down_or_output(&mut self, output: &Output) -> bool {
+        if let Some(monitor) = self.active_monitor() {
+            let workspace = monitor.active_workspace();
+
+            if !workspace.columns.is_empty() {
+                let column = &workspace.columns[workspace.active_column_idx];
+                let curr_idx = column.active_tile_idx;
+                let new_idx = min(column.active_tile_idx + 1, column.tiles.len() - 1);
+                if curr_idx != new_idx {
+                    workspace.focus_down();
+                    return false;
+                }
+            }
+        }
+
+        self.focus_output(output);
+        true
+    }
+
```

Comment:
You should add the two new actions to tests at the bottom of this file. At least into the Op struct which participates in randomized tests.

Reply:
Done, sorry I missed this I just saw It in the documentation :see_no_evil: